dev/run-pip-tests

```diff
@@ -120,6 +122,10 @@ for python in "${PYTHON_EXECS[@]}"; do
     python "$FWDIR"/dev/pip-sanity-check.py
     echo "Run the tests for context.py"
     python "$FWDIR"/python/pyspark/context.py
+    if [ -n "$TEST_PYARROW" ]; then
+      echo "Run tests for pyarrow"
+      SPARK_TESTING=1 "$FWDIR"/bin/pyspark pyspark.sql.tests ArrowTests
```

Comment:
This is just a temporary addition to make sure pyarrow tests run because they required a conda env, for now